src/Hateoas/Serializer/JsonHalSerializer.php

```diff
@@ -33,35 +35,40 @@ public function serializeLinks(array $links, JsonSerializationVisitor $visitor,
             }
         }
 
-        $visitor->addData('_links', $serializedLinks);
+        $visitor->visitProperty(new StaticPropertyMetadata(__CLASS__, '_links', $serializedLinks), $serializedLinks);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function serializeEmbeddeds(array $embeddeds, JsonSerializationVisitor $visitor, SerializationContext $context)
+    public function serializeEmbeddeds(array $embeddeds, SerializationVisitorInterface $visitor, SerializationContext $context)
     {
         $serializedEmbeddeds = array();
         $multiple = array();
+        $navigator = $context->getNavigator();
+
         foreach ($embeddeds as $embedded) {
             $context->pushPropertyMetadata($embedded->getMetadata());
+            try {
+                if (!isset($serializedEmbeddeds[$embedded->getRel()])) {
+                    $serializedEmbeddeds[$embedded->getRel()] = $navigator->accept($embedded->getData(), null, $context);
+                } elseif (!isset($multiple[$embedded->getRel()])) {
+                    $multiple[$embedded->getRel()] = true;
 
-            if (!isset($serializedEmbeddeds[$embedded->getRel()])) {
-                $serializedEmbeddeds[$embedded->getRel()] = $context->accept($embedded->getData());
-            } elseif (!isset($multiple[$embedded->getRel()])) {
-                $multiple[$embedded->getRel()] = true;
+                    $serializedEmbeddeds[$embedded->getRel()] = array(
+                        $serializedEmbeddeds[$embedded->getRel()],
+                        $navigator->accept($embedded->getData(), null, $context),
+                    );
+                } else {
+                    $serializedEmbeddeds[$embedded->getRel()][] = $navigator->accept($embedded->getData(), null, $context);
+                }
+            } catch (NotAcceptableException $e) {
```

Comment:
Would it make sense to log this error ?

Reply:
not really, in jms 2.0 `NotAcceptableException` is used to skip nulls , see https://github.com/schmittjoh/serializer/blob/ccb142ce8f1a3b034449e52238dee5bea07c18bb/src/JsonSerializationVisitor.php#L94